### PR TITLE
Javascript/javascript2/week2

### PR DESCRIPTION
**Pull request by aijazraja**

### CODE REVIEW

**Comment by NayanaKamtekar** [comment]
Could you please resolve conflicts so I can review your homework?

**Comment by aijazraja** [comment]
> Could you please resolve conflicts so I can review your homework?

Hi 
I reloved the conflict 
JS2 week2 is ready for cooments

File: javascript/javascript2/week2/index.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
...running_times": 7560},{"title": "Le divorce","year": 2003,"rating": 4.9,"votes": 10178,"running_times": 7020},{"title": "Le fabuleux destin d'Amélie Poulain","year": 2001,"rating": 8.3,"votes": 594927,"running_times": 7320},{"title": "Le grand bleu","year": 1988,"rating": 7.6,"votes": 41252,"running_times": 10080},{"title": "Le locataire","year": 1976,"rating": 7.8,"votes": 30919,"running_times": 7560},{"title": "Le mépris","year": 1963,"rating": 7.7,"votes": 22380,"running_times": 6180},{"title": "Le procès","year": 1962,"rating": 7.8,"votes": 15401,"running_times": 7140},{"title": "Le salaire de la peur","year": 1953,"rating": 8.2,"votes": 40345,"running_times": 9360},{"title": "Le violon rouge","year": 1998,"rating": 7.7,"votes": 28394,"running_times": 8400},{"title": "Lean on Me","year": 1989,"rating": 7.4,"votes": 15573,"running_times": 6480},{"title": "Leap Year","year": 2010,"rating": 6.5,"votes": 78668,"running_times": 6000},{"title": "Leap of Faith","year": 1992,"rating": 6.1,"votes": 10526,"running_times": 6480},{"title": "Leatherface: Texas Chainsaw Massacre III","year": 1990,"rating": 5.2,"votes": 11025,"running_times": 5100},{"title": "Leatherheads","year": 2008,"ratin
```
Comment (NayanaKamtekar) [inline javascript/javascript2/week2/index.js]: console log for moviesObject showing following object
 {averageMovies: 3837, badMovies: 88, goodMovies: 2601} 

Total movies are 6527 and your object shows total movies 6526.

Reduce return accumulator only once and first parameter is a accumulator.
```
let abc = moviesWithTag.reduce((tageCount,oneMovie)=>{
    if(oneMovie.tag=='Good'){
        tageCount.goodMovies++;
    }
    
    else if(oneMovie.tag=='Average'){
        tageCount.averageMovies++;
    }
    
    else if(oneMovie.tag='Bad'){
        tageCount.badMovies++;
    }
    return tageCount; 
},{goodMovies: 0, averageMovies: 0, badMovies: 0});
```

**Review by NayanaKamtekar — APPROVED** [review]
Good work.
There is scope for improvement in HYF app. It is good if both searching by product name and maximum price works together.

**Pull request merged**